portalnet/src/overlay.rs

```diff
@@ -482,11 +479,13 @@ where
     ) -> anyhow::Result<()> {
         match self.validator.validate_content(content_key, content).await {
             Ok(_) => {
-                self.metrics.report_validation(true);
+                self.metrics
+                    .report_validation(&self.protocol.to_string(), true);
```

Comment:
My update to the metrics strategy requires the addition of a `protocol` argument to the metrics reporting functions. I don't love this. I tried a couple ideas to eliminate this design, but I couldn't find something satisfactorily simple. This pr is already somewhat complex, and that would have increased the complexity in exchange for simply adding a `protocol` argument. I opted to leave the simple design for now. But, I'd be curious to hear thoughts from others. Is it worth trying to refactor the `protocol` argument away? Any neat ideas on the best way to accomplish this?

Reply:
Added `StorageMetricsReporter` & `OverlayMetricsReporter` to handle the per-subnetwork metrics reporting, avoiding the need to use an explicit `self.protocol` throughout `OverlayService` & `Storage`